### PR TITLE
Improve top-level commands; add reference links

### PR DESCRIPTION
**Pull request by hearvox**
Related: [/wp-cli/issues/4302](https://github.com/wp-cli/wp-cli/issues/4302)

Also updated README.md with the following new command descriptions.

Changes:
___
Improve 'wp comment' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/Comment_Command.php
(Related: (wp-cli/wp-cli#4302)[https://github.com/wp-cli/wp-cli/issues/4302]))

Current description: Manage comments.

New description:
Creates, updates, deletes, and moderates comments.

___
https://github.com/wp-cli/entity-command/issues
Improve 'wp menu' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/Menu_Command.php

New second sentence:
See the [Navigation Menus](https://developer.wordpress.org/themes/functionality/navigation-menus/) reference in the Theme Handbook.

___
Improve 'wp option' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/Option_Command.php

Current description: Manage options.

New description:
Retrieves and sets site options, including plugin and WordPress settings.

See the [Plugin Settings API](https://developer.wordpress.org/plugins/settings/settings-api/) and the 
[Theme Options](https://developer.wordpress.org/themes/customize-api/) for more information on adding customized options.

___
Improve 'wp post-type' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/Post_Type_Command.php
Current description: Manage post types.

New second sentence:
Get details on WordPress built-in and [custom post types](https://developer.wordpress.org/plugins/post-types/).

___
Improve 'wp site' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/Site_Command.php
Current description: Perform site-wide operations.

New description:
Performs site-wide operations on a multisite install.

___
Improve 'wp taxonomy' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/Taxonomy_Command.php 
Current description: Manage taxonomies.

New description:
Retrieves information about registered taxonomies.

See references for [built-in taxonomies](https://developer.wordpress.org/themes/basics/categories-tags-custom-taxonomies/) and [custom taxonomies](https://developer.wordpress.org/plugins/taxonomies/working-with-custom-taxonomies/).

___
Improve 'wp term' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/Term_Command.php
Current description: Manage terms.

New description:
Manages taxonomy terms and term meta, including create, delete, and list.

See reference for [taxonomies and their terms](https://codex.wordpress.org/Taxonomies).

___
Improve 'wp user' command description in:
https://github.com/wp-cli/entity-command/blob/master/src/User_Command.php
Current description: Manage users.

New description:
Manages users, along with their roles, capabilities, and meta.

See references for [Roles and Capabilities](https://codex.wordpress.org/Roles_and_Capabilities)and [WP User class](https://codex.wordpress.org/Class_Reference/WP_User).

### CODE REVIEW

File: src/User_Command.php [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <?php
 
 /**
- * Manage users.
+ * Manages users, along with their roles, capabilities, and meta.
+ *
+ * See references for [Roles and Capabilities](https://codex.wordpress.org/Roles_and_Capabilities)and [WP User class](https://codex.wordpress.org/Class_Reference/WP_User).
```
Comment (schlessera) [inline src/User_Command.php]: Missing a space after the first link.

File: src/Term_Command.php [hunk 2]
```diff
@@ -3,7 +3,9 @@
 use WP_CLI\Utils;
 
 /**
- * Manage terms.
+ * Manages taxonomy terms and term meta, including create, delete, and list.
```
Comment (schlessera) [inline src/Term_Command.php]: `including` would need to be followed by nouns.

Here's two options:

1. `Manages taxonomy terms and term meta, including creation, deletion, and enumeration`
2. `Manages taxonomy terms and term meta and allows to create, delete, and list them.`

File: src/Post_Type_Command.php [hunk 3]
```diff
@@ -2,6 +2,8 @@
 /**
  * Retrieves details on the site's registered post-types.
  *
+ * Get details on WordPress built-in and [custom post types](https://developer.wordpress.org/plugins/post-types/).
```
Comment (schlessera) [inline src/Post_Type_Command.php]: `on WordPress' built-in`

File: src/Term_Command.php (hunk 2 above)
Comment (hearvox) [reply src/Term_Command.php]: Tryig to keep it under 80chars, how about:
`Manages taxonomy terms and term meta, with create, delete, and list commands.`

File: src/Post_Type_Command.php [hunk 4]
```diff
@@ -1,6 +1,8 @@
 <?php
 /**
- * Retrieves details on the site's registered post-types.
+ * Retrieves details on the site's registered post types.
+ *
+ * Get information on the WordPress' built-in and the site's [custom post types](https://developer.wordpress.org/plugins/post-types/).
```
Comment (schlessera) [inline src/Post_Type_Command.php]: The first `the` (in front of `WordPress'`) seems grammatically wrong to me.

Comment (hearvox) [reply src/Post_Type_Command.php]: Sorry, that was a mistake. Correcting now.

Also, I don't know what this means:
> please re-run wp scaffold package-readme to update the README.md file again.

Is that something I need to do on my end? If so, please send details.

Comment (schlessera) [reply src/Post_Type_Command.php]: The `README.md` files inside of the command packages are automatically generated through the `wp scaffold package-readme` command: https://github.com/wp-cli/scaffold-package-command#wp-scaffold-package-readme

So, any time you change the documentation part of the source files, you'll need to re-run `wp scaffold package-readme . --force` from inside the root folder of that package to update the `README.md` file to the latest changes.

You might need to install the command first, if you haven't done so: `wp package install wp-cli/scaffold-package-command`

Comment (hearvox) [reply src/Post_Type_Command.php]: Ah, I see: But I'm editing files in my forked repo directly on GitHub, not in a local version on my computer. So I'm guessing there is no way to run that package here on GitHub. Is there another way to update the README without me having to download the repo?

Comment (schlessera) [reply src/Post_Type_Command.php]: We'll run the script over all packages once we've merged the updated descriptions.

Comment (hearvox) [reply src/Post_Type_Command.php]: Thanks, @schlessera , that will save me some time (tho cost you some). Appreciated. We got diverted but have a few more PRs to submit for command descriptions. Will try to get to that today. And thanks to @miya0001 for all his help too.
cc @ako80218

**Pull request merged**